code/game/objects/structures/mirror.dm

```diff
@@ -36,44 +73,149 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
+			return
+		beardresser.set_facial_hairstyle("Shaved", update = TRUE)
 
-	//handle normal hair
-	var/new_style = tgui_input_list(user, "Select a hairstyle", "Grooming", GLOB.hairstyles_list)
+/obj/structure/mirror/proc/change_hair(mob/living/carbon/human/hairdresser)
+	var/new_style = tgui_input_list(hairdresser, "Select a hairstyle", "Grooming", GLOB.hairstyles_list)
 	if(isnull(new_style))
 		return TRUE
-	if(!user.can_perform_action(src, FORBID_TELEKINESIS_REACH))
-		return TRUE //no tele-grooming
 	if(HAS_TRAIT(hairdresser, TRAIT_BALD))
-		to_chat(hairdresser, span_notice("If only growing back hair were that easy for you..."))
+		to_chat(hairdresser, span_notice("If only growing back hair were that easy for you... The reminder makes you feel terrible."))
+		hairdresser.add_mood_event("bald_hair_day", /datum/mood_event/bald)
 		return TRUE
 
 	hairdresser.set_hairstyle(new_style, update = TRUE)
 
-/obj/structure/mirror/examine_status(mob/user)
+/obj/structure/mirror/proc/change_name(mob/living/carbon/human/user)
+	var/newname = sanitize_name(tgui_input_text(user, "Who are we again?", "Name change", user.name, MAX_NAME_LEN), allow_numbers = TRUE) //It's magic so whatever.
+	if(!newname)
+		return TRUE
+	user.real_name = newname
+	user.name = newname
+	if(user.dna)
+		user.dna.real_name = newname
+	if(user.mind)
+		user.mind.name = newname
+
+// Erm ackshually the proper term is species. Get it right??
+/obj/structure/mirror/proc/change_race(mob/living/carbon/human/racist)
+	var/racechoice = tgui_input_list(racist, "What are we again?", "Race change", selectable_races)
+	if(isnull(racechoice))
+		return TRUE
+	if(!selectable_races[racechoice])
+		return TRUE
+
+	var/datum/species/newrace = selectable_races[racechoice]
+	racist.set_species(newrace, icon_update = FALSE)
+	if(HAS_TRAIT(racist, TRAIT_USES_SKINTONES))
+		var/new_s_tone = tgui_input_list(racist, "Choose your skin tone", "Race change", GLOB.skin_tones)
+		if(new_s_tone)
+			racist.skin_tone = new_s_tone
+			racist.dna.update_ui_block(DNA_SKIN_TONE_BLOCK)
+	else if(HAS_TRAIT(racist, TRAIT_MUTANT_COLORS) && !HAS_TRAIT(racist, TRAIT_FIXED_MUTANT_COLORS))
+		var/new_mutantcolor = input(racist, "Choose your skin color:", "Race change", racist.dna.features["mcolor"]) as color|null
+		if(new_mutantcolor)
+			var/temp_hsv = RGBtoHSV(new_mutantcolor)
+
+			if(ReadHSV(temp_hsv)[3] >= ReadHSV("#7F7F7F")[3]) // mutantcolors must be bright
+				racist.dna.features["mcolor"] = sanitize_hexcolor(new_mutantcolor)
+				racist.dna.update_uf_block(DNA_MUTANT_COLOR_BLOCK)
+
+			else
+				to_chat(racist, span_notice("Invalid color. Your color is not bright enough."))
+				return TRUE
+
+		racist.update_body(is_creating = TRUE)
+		racist.update_mutations_overlay() // no hulk lizard
+
+// possible Genders: MALE, FEMALE, PLURAL, NEUTER
+// possible Physique: MALE, FEMALE
+// saved you a click (many)
+/obj/structure/mirror/proc/change_sex(mob/living/carbon/human/sexist)
```

Comment:
probably `sex_changer` here as well. i wish we could keep the funny too but unfortunately we must larp as having standards for varnames instead of shitposting 24/7 (perhaps only a smidge of shitposting is okay at times)

Reply:
It's so over